_source/logzio_collections/_log-sources/mongo.md

```diff
@@ -0,0 +1,175 @@
+gem "fluent-plugin-logzio", "0.0.21"
+gem "fluent-plugin-record-modifier"
+```
+
+
+##### Install Fluentd and the Logz.io plugin
+
+
+```shell
+gem install fluentd fluent-plugin-logzio
+```
+
+##### Set up Fluentd
+
+```shell
+fluentd --setup ./fluent
+```
+
+##### Configure Fluentd with Logz.io output
+
+Add this code block to your Fluent configuration file (`fluent.conf` by default).
+
+See the configuration parameters below the code block.👇
+
+```conf
+# To ignore fluentd logs
+<label @FLUENT_LOG>
+  <match fluent.*>
+    @type null
+  </match>
+</label>
+# Tailing mongodb logs
+<source>
+  @type tail
+  @id mongodb_logs
+  path <<MONGODB-FILE-PATH>>
+  # If you're running on windows, change the pos_file to a Windows path
+  pos_file /var/log/fluentd-mongodb.log.pos
+  tag logzio.mongodb.*
+  read_from_head true
+  <parse>
+    @type json
+  </parse>
+</source>
+# Parsing the logs
+<filter logzio.mongodb.**>
+  @type record_modifier
+  <record>
+    type  mongodb-fluentd
+    message ${record["msg"]}
+    mongodb_timestamp ${record["t"]["$date"]}
+    log_id ${record["id"].to_s}
+  </record>
+  remove_keys msg,t,id
+</filter>
+# Sending logs to Logz.io
+<match logzio.mongodb.**>
+  @type logzio_buffered
+  endpoint_url https://listener.logz.io:8071?token=<<LOGZIO-SHIPPING-TOKEN>>
+  output_include_time true
+  output_include_tags true
+  http_idle_timeout 10
+  <buffer>
+      @type memory
+      flush_thread_count 4
+      flush_interval 3s
+      chunk_limit_size 16m      # Logz.io bulk limit is decoupled from chunk_limit_size. Set whatever you want.
+      queue_limit_length 4096
+  </buffer>
+</match>
+```
+
+###### Parameters
+
+| Parameter | Description |
+|---|---|
+| `<<MONGODB-FILE-PATH>>` | Path to the log file of your MongoDB. |
+| endpoint_url | A url composed of your Logz.io region's listener URL, account token, and log type. {% include log-shipping/listener-var.html %} {% include log-shipping/log-shipping-token.html %} |
+| output_include_time | To add a timestamp to your logs when they're processed, `true` (recommended). Otherwise, `false`. |
+| output_include_tags | To add the `fluentd` tag to logs, `true`. Otherwise, `false`. If `true`, use in combination with `output_tags_fieldname`. |
+| output_tags_fieldname | If `output_include_tags` is `true`, sets output tag's field name. The default is `fluentd_tag` |
+| http_idle_timeout | Time, in seconds, that the HTTP connection will stay open without traffic before timing out. |
+
+
+
+##### Allow Fluentd to read from the MongoDB log file
+
+Allow Fluend to read from the MongoDB log file by setting the read access as follows:
+
+###### On macOS and Linux
+
+```shell
+sudo chmod 604 <<PATH-TO-LOG-FILE>>
+```
+
+* Replace `<<PATH-TO-LOG-FILE>>` with the path to the MongoDB log file.
+
+
+###### On Windows
+
+Enable the read access in the file properties.
+
+
+
+##### Run Fluentd
+
+```shell
+fluentd -c ./fluent.conf --gemfile <<PATH-TO-GEMFILE>>/Gemfile
```

Comment:
Since we instruct them to create it under the fluent directory, we can change the command to:
fluentd -c ./fluent.conf --gemfile ./Gemfile